onecodex/analyses.py

```diff
@@ -239,7 +239,56 @@ def _metadata_fetch(self, metadata_fields, label=None):
 
         return magic_metadata, magic_fields
 
-    def to_df(
+    def to_df(self, analysis_type=AnalysisType.Classification, **kwargs):
+        """
+        Transform Analyses of samples in a `SampleCollection` into tabular format.
+
+        Parameters
+        ----------
+        analysis_type : {'classification', 'functional'}, optional
+            The `analysis_type` to aggregate, corresponding to AnalysisJob.analysis_type
+        kwargs : dict, optional
+             Keyword arguments specific to the `analysis_type`; see each individual function definition
+        """
+        generate_df = {
+            AnalysisType.Classification: self._to_classification_df,
+            AnalysisType.Functional: self._to_functional_df,
+        }
+        return generate_df[AnalysisType(analysis_type)](**kwargs)
+
+    def _to_functional_df(
+        self,
+        annotation=FunctionalAnnotations.Pathways,
+        taxa_stratified=True,
+        metric="coverage",
```

Comment:
can you make `metric` an enum too? I'm liking the free validation we get with the other enums you added 👍

Reply:
done